src/mds/FSMap.h

```diff
@@ -259,17 +259,21 @@ class FSMap {
       return;
     }
 
-    for (auto &f : filesystems) {
-      std::string_view fs_name = f.second->mds_map.get_fs_name();
+    for(auto it = filesystems.begin(); it != filesystems.end();) {
+      std::string_view fs_name = (*it).second->mds_map.get_fs_name();
       if (std::find(allowed.begin(), allowed.end(), fs_name) == allowed.end()) {
-	filesystems.erase(f.first);
+        it = filesystems.erase(it);
```

Comment:
Since the issue is not immediately apparent, I think it's a good idea to add a XXX note here (or at the beginning of the loop) to mention why we switched the looping style. This way any future modifier doesn't have to re-discover it.

Reply:
Hi @rishabh-d-dave,

soon after the move to the C++20 is done the `std::erase_if` could be assumed as a fix.
A fix in the `std::vector<mds_gid_t> FSMap::stop(mds_gid_t who)` will be done with anonther PR (integration test again segfaults but at the fs:multifs).